vms/job/tests/test_jobDetails.py

```diff
@@ -11,53 +10,183 @@
+        created_event = JobDetails.register_valid_event()
+        self.job_details_page.navigate_to_event_list_view()
+
+        job_details_page = self.job_details_page
+        job_details_page.live_server_url = self.live_server_url
+
+        # Create empty job
+        job = [created_event.id, '', '', '', '']
+        job_details_page.navigate_to_job_list_view()
+        job_details_page.go_to_create_job_page()
+        job_details_page.fill_job_form(job)
+
+        # Check error messages
+        self.check_error_messages()
+
+    def test_invalid_job_edit(self):
+        created_event = JobDetails.register_valid_event()
+        self.job_details_page.navigate_to_event_list_view()
+        JobDetails.register_valid_job(created_event)
+
+        job_details_page = self.job_details_page
+        job_details_page.live_server_url = self.live_server_url
+        job_details_page.navigate_to_job_list_view()
+        job_details_page.go_to_edit_job_page()
+
+        null_valued_job = [created_event.id, '', '', '', '']
+        job_details_page.fill_job_form(null_valued_job)
+        self.check_error_messages()
+
+    def test_valid_job_edit(self):
+        created_event = JobDetails.register_valid_event()
+        self.job_details_page.navigate_to_event_list_view()
+        JobDetails.register_valid_job(created_event)
+
+        job_details_page = self.job_details_page
+        job_details_page.live_server_url = self.live_server_url
+        job_details_page.navigate_to_job_list_view()
+
+        edit_job = ['event', 'new job name', 'new-job-description', '2050-06-16', '2050-06-16']
+        job_details_page.go_to_edit_job_page()
+        job_details_page.fill_job_form(edit_job)
+        job_details_page.navigate_to_job_list_view()
+
+        self.assertEqual(job_details_page.get_name(), edit_job[1])
+        self.assertEqual(job_details_page.get_description(), edit_job[2])
+
+    def test_job_delete(self):
+        job_details_page = self.job_details_page
+        job_details_page.live_server_url = self.live_server_url
+
+        created_event = JobDetails.register_valid_event()
+        job_details_page.navigate_to_event_list_view()
+        JobDetails.register_valid_job(created_event)
+
+        job_details_page.navigate_to_job_list_view()
+        self.assertEqual(job_details_page.get_delete_element('').text, 'Delete')
+        job_details_page.get_delete_element('//a').click()
+        self.assertEqual(job_details_page.get_deletion_context(), 'Delete Job')
+        job_details_page.submit_form()
+
+    def test_create_job_with_no_event_present(self):
+        job_details_page = self.job_details_page
+        job_details_page.navigate_to_event_list_view()
+        job_details_page.click_link(job_details_page.jobs_tab)
+        self.assertEqual(job_details_page.remove_i18n(self.driver.current_url),
+                         self.live_server_url + job_details_page.job_list_page)
+        self.assertEqual(job_details_page.get_message_context(),
```

Comment:
Make this a variable inside `JobDetailsPage` itself (multiple uses might be possible.)